pkg/storage/etcd/v3_helper.go

```diff
@@ -34,6 +33,7 @@ import (
 
 var (
 	ErrNotFound = errors.New("etcd: key not found")
+	ErrTestFail = errors.New("etcd: test failed")
```

Comment:
Let me reply to all your comments w.r.t. these errors.

What's the problem?
Those are etcd errors and would be handled specially. See [here](https://github.com/kubernetes/kubernetes/blob/c505a5d49d35011abcb9c4fa853eb33b7b1efb6e/pkg/storage/etcd/util/etcd_util.go#L29-L65).
However, etcd3 wouldn't have return those errors.

How to resolve them?
The simplest way is to make etcd2 errors and return them. That's my plan. What's your opinion?

/cc @xiang90